p2p/starknet/handlers.go

```diff
@@ -134,16 +136,45 @@ func (h *Handler) onBlockBodiesRequest(req *spec.BlockBodiesRequest) (Stream[pro
 }
 
 func (h *Handler) onEventsRequest(req *spec.EventsRequest) (Stream[proto.Message], error) {
-	// todo: read from bcReader and adapt to p2p type
-	count := uint64(0)
+	it, err := h.newIterator(req.Iteration)
+	if err != nil {
+		return nil, err
+	}
+
+	fin := func(it *iterator) *spec.EventsResponse {
+		return &spec.EventsResponse{
+			Id: &spec.BlockID{
+				Number: it.BlockNumber(),
+			},
+			Responses: &spec.EventsResponse_Fin{
+				Fin: nil,
+			},
+		}
+	}
+
 	return func() (proto.Message, bool) {
-		if count > 3 {
-			return nil, false
+		if !it.Valid() {
+			return fin(it), false
 		}
-		count++
+
+		block, err := it.Block()
+		if err != nil {
+			h.log.Errorw("Failed to fetch block", "err", err)
+			return fin(it), false
+		}
+		it.Next()
+
+		var events []*core.Event
+		for _, item := range block.Receipts {
+			events = append(events, item.Events...)
+		}
+
 		return &spec.EventsResponse{
-			Id: &spec.BlockID{
-				Number: count - 1,
+			Id: core2p2p.AdaptBlockID(block.Header),
+			Responses: &spec.EventsResponse_Events{
+				Events: &spec.Events{
+					Items: utils.Map(events, core2p2p.AdaptEvent),
```

Comment:
We need to chunk the events in case the messages exceeds 1MB, but we can leave it for later.